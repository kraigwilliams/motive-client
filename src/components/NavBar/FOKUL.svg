<?xml version="1.0" encoding="utf-8"?>
<!-- Generator: Adobe Illustrator 23.1.1, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
<svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
	 viewBox="0 0 556 184.2" style="enable-background:new 0 0 556 184.2;" xml:space="preserve">
<g>
	<path d="M110.3,116.3c1.1-27.9,11.7-50.7,36.1-66.2c19-12,48.3-11.2,65.3,3.5c13.9,12,18.8,27.9,15.9,46c-0.7,4.4-1.9,8.8-2.9,13.2
		c-4.6,21.4-17.3,39.5-34.9,52.1c-25,17.8-69.6,6.8-77.9-31.1C110.8,128.9,110.3,121.3,110.3,116.3z M128.1,115.8
		c0.2,4.2,0.4,11.2,1.5,15.4c5.6,22.2,31.4,32.1,50.6,19.4c14.7-9.7,24.8-25.3,28.1-42.8c0.7-3.8,1.7-7.6,2.3-11.4
		c4-21.1-10-37.3-30.4-39.4c-9.7,0.2-18.8,2.1-26.9,8.1C136.1,77.8,129.2,95.3,128.1,115.8z"/>
	<path d="M53.1,58.7c-3.7,16.8-4.4,20.8-9.5,38.3c1.9,2.1,22.3,0.3,30.6,0c2.1-0.1,4.3,0,6.3,0.4c4.1,0.8,6.6,3.8,6.6,7.5
		c0.1,3.8-2.2,6.6-6.3,7.8c-2.3,0.7-4.6,0.5-7,0.5c-11.3,0-22.5,0-33.8,0.4c-1.7,1.5-1.7,3.3-2.2,4.9c-4.2,14.8-8.3,29.5-11.9,44.4
		c-0.4,1.9-1,3.7-1.9,5.4c-2,3.6-5.5,5.1-9.1,4.3c-3.6-0.8-6-3.8-6.2-7.9c-0.2-3.2,0.5-6.4,1.2-9.5c3.5-14.9,7.3-29.8,11.4-44.5
		c5.8-20.3,13.6-59.3,16.2-65c2.2-5,11.5-3.8,13.6-3.9c20-0.9,32,0.9,52,0c1.9-0.1,3.9-0.1,5.7,0.2c4.7,0.7,7.7,3.9,7.7,8
		c0,4.1-2.8,7.3-7.5,8.2c-2.3,0.4-4.7,0.5-7,0.5c-17.2,0.3-26.7-0.7-43.9-0.4C56.8,58.4,55.3,58.5,53.1,58.7z"/>
	<g>
		<path d="M356.8,43.7c-5.6-4.5-8.8-0.2-14.3,2.6c-1,0.5-12.8,9.7-17.4,13.3c-15.8,12.5-33.4,27-49.6,38.8c-2,1.4-3.7,3.3-6.4,3.8
			c-1-2-0.1-3.3,0.3-4.6c4.1-14.5,8.2-29,12.4-43.4c1.6-5.7-1-11.5-6.2-13.8c-4.4-1.9-8.5,0.2-10.9,5.9c-0.7,1.7-1.2,3.5-1.8,5.3
			c-1.8,6.1-3.4,12.2-5.3,18.3c-2.8,9.2-14.9,59.1-19.7,74.3c-1.3,4.2-2.4,8.6-3.6,12.9c-0.5,1.8-0.9,3.7-1.1,5.6
			c-0.4,5.4,1.9,9.3,6,10.3c3.9,1,7.4-1,9.4-5.6c0.7-1.5,1.1-3.1,1.5-4.7c3-10.2,5.9-20.3,8.9-30.5c0.7-2.2,0.9-4.9,3.1-6
			c7.7-3.7,14.7-8.6,21.9-13.2c0.7,1.1,1.2,1.6,1.5,2.3c6.4,17.7,12.3,31.5,19.9,48.5c1,2.3,2.1,4.7,3.6,6.8
			c2.5,3.7,6.4,4.8,9.7,3.1s4.9-5.6,3.9-10.2c-0.5-2.3-1.4-4.5-2.4-6.6c-5-10.5-7.5-18.3-12.3-28.9c-3.6-8-7.5-15.9-10.1-23.8
			c1-2.2,2.2-2.8,3.3-3.5c16.2-10.9,36.6-29.2,50.5-42.8C356.2,53.7,360.5,46.7,356.8,43.7z"/>
		<path d="M467.8,41.4c-3.1-0.9-6.3,0.4-8.5,3.6c-1.6,2.3-2.3,5.1-3.2,7.7c-2.6,7.5-17.2,60.3-23.9,72.8
			c-6.2,11.6-13.6,25.3-25.9,29.1c-4.4,1.4-8.2,1.8-12.8,0.1l-6.5-3.5c-7.4-5.4-11.6-14.2-11.4-24.8c0.2-11.4,11.8-57.6,15.4-69.4
			c0.6-2,1.3-4.1,1.5-6.2c0.3-3.6-1.1-6.5-4.2-8.1c-3.1-1.6-5.9-0.7-8.3,1.9c-1.6,1.7-2.5,3.8-3.2,6.1
			c-4.4,12.9-18.2,63.6-18.8,77.5c0.2,2.1,0.3,5.2,0.7,8.2c2.1,14.2,10.3,23.5,20.6,31.2c10.4,7.8,24.9,6.5,34.7,3.6
			c4-1.2,11.6-7.5,14.6-10.6c10.9-11.4,14.5-16.6,20.6-30.5c8.9-20.3,16.4-51.9,23.1-73.2c0.7-2.2,1.4-4.5,1.2-6.9
			C473.4,45.7,471.1,42.3,467.8,41.4z"/>
		<path d="M543.3,165c-0.4-3.1-2.2-5-5.1-6c-3-1-6.2-1.3-9.5-1.4c-8.3-0.3-24.6-0.6-32.9-0.9c-1.7-0.1-3.5,0.2-5.6-0.6
			c0,0,22.2-84.2,26.8-97.9c1.3-3.8,2.8-7.7,2.5-11.8c-0.2-2.8-3.5-4.5-6.2-5.5c-2.6-0.9-5-0.3-7.1,1.7c-1.6,1.5-2.5,3.5-3.3,5.4
			c-12.9,30.9-32,108.7-32.2,110.2c-0.6,6.2,1.8,10.2,7.6,12.5c3.4,1.4,7,2.1,10.6,2.2c11.7,0.5,31.4,0.8,43.1,1.2
			c2.6,0.1,5.1-0.2,7.4-1.6C542.3,170.8,543.7,168.3,543.3,165z"/>
	</g>
</g>
</svg>
