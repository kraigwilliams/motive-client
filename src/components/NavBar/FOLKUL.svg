<?xml version="1.0" encoding="utf-8"?>
<!-- Generator: Adobe Illustrator 23.1.1, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
<svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
	 viewBox="0 0 556 184.2" style="enable-background:new 0 0 556 184.2;" xml:space="preserve">
<g>
	<path d="M96,96.1c0.9-24,10.1-43.6,31-56.9c16.3-10.3,41.5-9.6,56.1,3c11.9,10.3,16.2,24,13.7,39.5c-0.6,3.8-1.6,7.6-2.5,11.3
		c-4,18.4-14.9,33.9-30,44.8c-21.5,15.3-59.8,5.8-66.9-26.7C96.4,106.9,96,100.4,96,96.1z M111.3,95.6c0.2,3.6,0.3,9.6,1.3,13.2
		c4.8,19.1,27,27.6,43.5,16.7c12.6-8.3,21.3-21.7,24.1-36.8c0.6-3.3,1.5-6.5,2-9.8c3.4-18.1-8.6-32.1-26.1-33.9
		c-8.3,0.2-16.2,1.8-23.1,7C118.2,63,112.2,78,111.3,95.6z"/>
	<path d="M46.8,46.6c-3.2,14.4-3.8,17.9-8.2,32.9c1.6,1.8,19.2,0.3,26.3,0c1.8-0.1,3.7,0,5.4,0.3c3.5,0.7,5.7,3.3,5.7,6.4
		c0.1,3.3-1.9,5.7-5.4,6.7c-2,0.6-4,0.4-6,0.4c-9.7,0-19.3,0-29,0.3c-1.5,1.3-1.5,2.8-1.9,4.2c-3.6,12.7-7.1,25.3-10.2,38.2
		c-0.3,1.6-0.9,3.2-1.6,4.6c-1.7,3.1-4.7,4.4-7.8,3.7c-3.1-0.7-5.2-3.3-5.3-6.8c-0.2-2.7,0.4-5.5,1-8.2c3-12.8,6.3-25.6,9.8-38.2
		c5-17.4,11.7-51,13.9-55.9c1.9-4.3,9.9-3.3,11.7-3.4c17.2-0.8,27.5,0.8,44.7,0c1.6-0.1,3.4-0.1,4.9,0.2c4,0.6,6.6,3.4,6.6,6.9
		s-2.4,6.3-6.4,7c-2,0.3-4,0.4-6,0.4c-14.8,0.3-22.9-0.6-37.7-0.3C50,46.3,48.7,46.4,46.8,46.6z"/>
	<g>
		<path d="M387.4,33.7c-4.8-3.9-7.6-0.2-12.3,2.2c-0.9,0.4-11,8.3-15,11.4c-13.6,10.7-28.7,23.2-42.6,33.3c-1.7,1.2-3.2,2.8-5.5,3.3
			c-0.9-1.7-0.1-2.8,0.3-4c3.5-12.5,7-24.9,10.7-37.3c1.4-4.9-0.9-9.9-5.3-11.9c-3.8-1.6-7.3,0.2-9.4,5.1c-0.6,1.5-1,3-1.5,4.6
			c-1.5,5.2-2.9,10.5-4.6,15.7c-2.4,7.9-12.8,50.8-16.9,63.8c-1.1,3.6-2.1,7.4-3.1,11.1c-0.4,1.5-0.8,3.2-0.9,4.8
			c-0.3,4.6,1.6,8,5.2,8.9c3.4,0.9,6.4-0.9,8.1-4.8c0.6-1.3,0.9-2.7,1.3-4c2.6-8.8,5.1-17.4,7.6-26.2c0.6-1.9,0.8-4.2,2.7-5.2
			c6.6-3.2,12.6-7.4,18.8-11.3c0.6,0.9,1,1.4,1.3,2c5.5,15.2,10.6,27.1,17.1,41.7c0.9,2,1.8,4,3.1,5.8c2.1,3.2,5.5,4.1,8.3,2.7
			s4.2-4.8,3.4-8.8c-0.4-2-1.2-3.9-2.1-5.7c-4.3-9-6.4-15.7-10.6-24.8c-3.1-6.9-6.4-13.7-8.7-20.5c0.9-1.9,1.9-2.4,2.8-3
			c13.9-9.4,31.4-25.1,43.4-36.8C386.9,42.3,390.6,36.3,387.4,33.7z"/>
		<path d="M482.8,31.7c-2.7-0.8-5.4,0.3-7.3,3.1c-1.4,2-2,4.4-2.7,6.6c-2.2,6.4-14.8,51.8-20.5,62.6c-5.3,10-11.7,21.7-22.3,25
			c-3.8,1.2-7,1.5-11,0.1l-5.6-3c-6.4-4.6-10-12.2-9.8-21.3c0.2-9.8,10.1-49.5,13.2-59.6c0.5-1.7,1.1-3.5,1.3-5.3
			c0.3-3.1-0.9-5.6-3.6-7c-2.7-1.4-5.1-0.6-7.1,1.6c-1.4,1.5-2.1,3.3-2.7,5.2c-3.8,11.1-15.6,54.7-16.2,66.6c0.2,1.8,0.3,4.5,0.6,7
			c1.8,12.2,8.9,20.2,17.7,26.8c8.9,6.7,21.4,5.6,29.8,3.1c3.4-1,10-6.4,12.5-9.1c9.4-9.8,12.5-14.3,17.7-26.2
			c7.6-17.4,14.1-44.6,19.8-62.9c0.6-1.9,1.2-3.9,1-5.9C487.6,35.4,485.6,32.5,482.8,31.7z"/>
		<path d="M547.6,137.9c-0.3-2.7-1.9-4.3-4.4-5.2c-2.6-0.9-5.3-1.1-8.2-1.2c-7.1-0.3-21.1-0.5-28.3-0.8c-1.5-0.1-3,0.2-4.8-0.5
			c0,0,19.1-72.4,23-84.1c1.1-3.3,2.4-6.6,2.1-10.1c-0.2-2.4-3-3.9-5.3-4.7c-2.2-0.8-4.3-0.3-6.1,1.5c-1.4,1.3-2.1,3-2.8,4.6
			c-11.1,26.6-27.5,93.4-27.7,94.7c-0.5,5.3,1.5,8.8,6.5,10.7c2.9,1.2,6,1.8,9.1,1.9c10.1,0.4,27,0.7,37,1c2.2,0.1,4.4-0.2,6.4-1.4
			C546.8,142.9,548,140.8,547.6,137.9z"/>
		<path d="M268.8,137.9c-0.3-2.7-1.9-4.3-4.4-5.2c-2.6-0.9-5.3-1.1-8.2-1.2c-7.1-0.3-21.1-0.5-28.3-0.8c-1.5-0.1-3,0.2-4.8-0.5
			c0,0,19.1-72.4,23-84.1c1.1-3.3,2.4-6.6,2.1-10.1c-0.2-2.4-3-3.9-5.3-4.7c-2.2-0.8-4.3-0.3-6.1,1.5c-1.4,1.3-2.1,3-2.8,4.6
			c-11.1,26.6-27.5,93.4-27.7,94.7c-0.5,5.3,1.5,8.8,6.5,10.7c2.9,1.2,6,1.8,9.1,1.9c10.1,0.4,27,0.7,37,1c2.2,0.1,4.4-0.2,6.4-1.4
			C267.9,142.9,269.1,140.8,268.8,137.9z"/>
	</g>
</g>
</svg>
