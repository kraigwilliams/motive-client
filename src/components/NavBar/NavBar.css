
.logo:hover {
    cursor: pointer;
    color: teal;
}
