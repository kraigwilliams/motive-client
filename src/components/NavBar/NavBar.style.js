import styled from 'styled-components'
import { colors } from '../constants'
import {Link} from 'react-router-dom'

export const NavWrapper = styled.header`
  background-color: ${colors.darkgrey};
  display: flex;
  flex-direction: row;
  padding: 10px 20px;
  justify-content: space-between;
  overflow: hidden;
  .logged-in {
    display: flex;
    flex-direction: row;
    height: 40px;
    width: fit-content;
    justify-items: right;
  }
`

export const Logo = styled(Link)`
  font-size: 20px;
  color: ${colors.white};
  text-decoration: none;
  font-weight: bolder;
  align-self: center;
  :hover{
    cursor: pointer;
    color: ${colors.teal}
  }
`;

export const MenuLink = styled(Link)`
  font-size: 16px;
  color: ${colors.white};
  text-decoration: none;
  margin-left: 25px;
  align-self: center;

  :hover{
    cursor: pointer;
    color: ${colors.coral}
  }
`;

export const MobileButton = styled.button`
  background: transparent;
  border: none;
  display: none;

  @media screen and (max-width: 768px) {
    display: block;
  }

  :hover {
    cursor: pointer;
  }
`;