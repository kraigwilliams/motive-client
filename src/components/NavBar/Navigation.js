import React from 'react';
import {Link} from 'react-router-dom'
import styled from 'styled-components'
import { colors } from '../constants'


const NavLinks = styled.ul`
  list-style: none;
  color: ${props => props.color ? props.color : colors.white};
  align-items: center;
  justify-content:flex-end;
  margin-right: 0px;
  display: flex;
  flex-direction: ${props => props.direction ? props.direction : 'column'};
  padding: 0;

`;

const NavItem = styled(Link)`
  color: ${props => props.color ? props.color : colors.white};
  text-decoration: none;
  margin-right: 0px;
  font-size: 16px;
  margin-left: ${props => props ? props.marginLeft : '0px'};
  :hover{
    cursor: pointer;
    color: ${colors.coral}
  }
`;

const Navigation = (props) => {
  return (
    <NavLinks direction={props.direction}>
        {/* <li className='nav-links' >
          <NavItem to='/' color={props.color}>
            About
          </NavItem>
        </li>
         */}

        <li className='nav-links' >
          <NavItem to='/dashboard' color={props.color}>
            Dashboard
          </NavItem>
        </li>

        {/* <li className='nav-links' color={props.color}>
          <NavItem to='/connections' color={props.color}>
            Connections
          </NavItem>
        </li> */}
      </NavLinks>
  )
}

export default Navigation;