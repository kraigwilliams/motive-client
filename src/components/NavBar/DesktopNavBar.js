import React from 'react'
import Navigation from './Navigation'
import styled from 'styled-components';
import { colors } from '../constants'

const DesktopNav = styled.nav`
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
  align-items: center;
  background: ${colors.darkgrey};
  color: ${colors.white};

  .nav-links{
    margin-left: 20px;
  @media screen and (max-width: 768px) {
      display: none;
    }
  }
  
  .ul {
    display:flex;
    flex-direction: row;
  }
`;

// const MobileButton = styled.button`
//   background: transparent;
//   border: none;
//   display: none;

//   @media screen and (max-width: 768px) {
//     display: block;
//   }
// `;

const DesktopNavBar = (props) => {
  return (
    <DesktopNav>
      
      <Navigation direction='row' marginLeft='25px'/>

      {/* <MobileButton type='button' onClick={props.toggleMobileNav}>
        Icon
      </MobileButton> */}
    </DesktopNav>
  )
}

export default DesktopNavBar;