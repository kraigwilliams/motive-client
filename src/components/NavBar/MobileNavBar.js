import React from "react";
import styled from "styled-components";
import { colors } from "../constants";
import Navigation from "./Navigation";

const MobileNav = styled.nav`
  width: fit-content;
  margin: 0px;
  padding: 10px;
  background: ${colors.blue};
  color: ${colors.darkergrey};
  align-self: flex-end;
  position: absolute;
  top: 60px;
  right: 0;

  @media screen and (min-width: 768px) {
    display: none;
  }
`;

const MobileNavBar = (props) => {
  return (
    <MobileNav
      onClick={props.toggleNav}
      displayMobileNavBar={props.displayMobileNavBar}
    >
      <Navigation
        direction="column"
        color={colors.white}
        linkmargin="50px 0px"
      />
    </MobileNav>
  );
};

export default MobileNavBar;
