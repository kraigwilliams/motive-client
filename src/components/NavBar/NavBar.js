import React, { Component } from 'react';
import DesktopNavBar from './DesktopNavBar';
import MobileNavBar from './MobileNavBar';
import TokenService from '../../services/token-service';
import UserContext from '../../contexts/UserContext.js';
import Burger from './Burger';
import {
  NavWrapper,
  Logo,
  UserLogo,
  MenuLink,
  MobileButton,
} from './NavBar.style';
import { ReactComponent as FokulLogo } from './FOLKUL_1.svg';

class NavBar extends Component {
  static contextType = UserContext;
  constructor(props) {
    super(props);
    this.state = {
      mobileNav: false,
    };
  }

  toggleMobileNavBar = () => {
    this.setState({
      mobileNav: !this.state.mobileNav,
    });
  };

  handleLogoutClick = () => {
    this.context.processLogout();
  };

  renderLogoutLink() {
    return (
      <div className="logged-in">
        <DesktopNavBar />

        <MenuLink onClick={this.handleLogoutClick} to="/login">
          Logout
        </MenuLink>

        <MobileButton type="button" onClick={this.toggleMobileNavBar}>
          <Burger open={this.state.mobileNav} />
        </MobileButton>
      </div>
    );
  }

  renderLoginLink() {
    return (
      <div className="logged-out" style={{ alignSelf: 'center' }}>
        <MenuLink to="/login">Login</MenuLink>{' '}
        <MenuLink to="/signup">Sign up</MenuLink>
      </div>
    );
  }

  render() {
    return (
      <>
        <NavWrapper>
          <Logo to="/">
            {TokenService.hasAuthToken() ? (
              <UserLogo username={this.context.user.firstname} />
            ) : (
              <FokulLogo
                aria-label="Folkul logo, click to go to landing page"
                className="logo"
                style={{ fill: 'white', width: '100px' }}
              />
            )}
            {/* <FokulLogo
              aria-label="Folkul logo, click to go to landing page"
              className="logo"
              style={{ fill: "white", width: "100px" }}
            /> */}
          </Logo>

          {TokenService.hasAuthToken()
            ? this.renderLogoutLink()
            : this.renderLoginLink()}

          {this.state.mobileNav && TokenService.hasAuthToken() ? (
            <MobileNavBar
              toggleNav={this.toggleMobileNavBar}
              displayMobileNavBar={this.state.mobileNav}
            />
          ) : null}
        </NavWrapper>
      </>
    );
  }
}

export default NavBar;
