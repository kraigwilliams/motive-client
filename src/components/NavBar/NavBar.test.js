import React from 'react';
import { shallow, mount, render } from 'enzyme';
// import MobileNavBar from './MobileNavBar'
import { NavWrapper } from './NavBar.style'
import renderer from 'react-test-renderer';
// import { ContentProvider } from './contexts/ContentContext'

describe(`<NavBar />`, () => {
  it('renders UI as expected', () => {
    const tree = renderer 
      .create(<NavWrapper />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})