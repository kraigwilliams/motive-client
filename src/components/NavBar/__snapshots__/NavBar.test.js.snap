// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> renders UI as expected 1`] = `
<header
  className="sc-AxirZ cMwKkI"
/>
`;
